docs/sql-ref-literals.md

```diff
@@ -0,0 +1,506 @@
+ * [Interval Literal](#interval-literal)
+
+### String Literal
+
+A string literal is used to specify a character string value.
+
+#### Syntax
+
+{% highlight sql %}
+'c [ ... ]' | "c [ ... ]"
+{% endhighlight %}
+
+#### <em>Parameters</em>
+
+<dl>
+  <dt><code><em>c</em></code></dt>
+  <dd>
+    One character from the character set. Use <code>\</code> to escape special characters (e.g., <code>'</code> or <code>\</code>).
+  </dd>
+</dl>
+
+#### <em>Examples</em>
+
+{% highlight sql %}
+SELECT 'Hello, World!' AS col;
+  +-------------+
+  |          col|
+  +-------------+
+  |Hello, World!|
+  +-------------+
+
+SELECT "SPARK SQL" AS col;
+  +---------+
+  |      col|
+  +---------+
+  |Spark SQL|
+  +---------+
+
+SELECT SELECT 'it\'s $10.' AS col;
+  +---------+
+  |      col|
+  +---------+
+  |It's $10.|
+  +---------+
+{% endhighlight %}
+
+### Null Literal
+
+A null literal is used to specify a null value.
+
+#### Syntax
+
+{% highlight sql %}
+NULL
+{% endhighlight %}
+
+#### Examples
+
+{% highlight sql %}
+SELECT NULL AS col;
+  +----+
+  | col|
+  +----+
+  |NULL|
+  +----+
+{% endhighlight %}
+
+### Boolean Literal
+
+A boolean literal is used to specify a boolean value.
+
+#### Syntax
+
+{% highlight sql %}
+TRUE | FALSE
+{% endhighlight %}
+
+#### Examples
+
+{% highlight sql %}
+SELECT TRUE AS col;
+  +----+
+  | col|
+  +----+
+  |true|
+  +----+
+{% endhighlight %}
+
+### Numeric Literal
+
+A numeric literal is used to specify a fixed or floating-point number.
+
+#### Integer Literal
+
+#### Syntax
+
+{% highlight sql %}
+[ + | - ] digit [ ... ] [ L | S | Y ]
+{% endhighlight %}
+
+#### Parameters
+
+<dl>
+  <dt><code><em>digit</em></code></dt>
+  <dd>
+    Any numeral from 0 to 9.
+  </dd>
+</dl>
+<dl>
+  <dt><code><em>L</em></code></dt>
+  <dd>
+    Case insensitive, indicates <code>BIGINT</code>, which is a 8-byte signed integer number.
+  </dd>
+</dl>
+<dl>
+  <dt><code><em>S</em></code></dt>
+  <dd>
+    Case insensitive, indicates <code>SMALLINT</code>, which is a 2-byte signed integer number.
+  </dd>
+</dl>
+<dl>
+  <dt><code><em>Y</em></code></dt>
+  <dd>
+    Case insensitive, indicates <code>TINYINT</code>, which is a 1-byte signed integer number.
+  </dd>
+</dl>
+<dl>
+  <dt><code><em>default (no postfix)</em></code></dt>
+  <dd>
+    Indicates a 4-byte signed integer number.
+  </dd>
+</dl>
+
+#### Examples
+
+{% highlight sql %}
+SELECT -2147483648 AS col;
+  +-----------+
+  |        col|
+  +-----------+
+  |-2147483648|
+  +-----------+
+
+SELECT 9223372036854775807l AS col;
+  +-------------------+
+  |                col|
+  +-------------------+
+  |9223372036854775807|
+  +-------------------+
+
+SELECT -32Y AS col;
+  +---+
+  |col|
+  +---+
+  |-32|
+  +---+
+
+SELECT 482S AS col;
+  +---+
+  |col|
+  +---+
+  |482|
+  +---+
+{% endhighlight %}
+
+#### Decimal Literal
+
+#### Syntax
+
+{% highlight sql %}
+[ + | - ] { digit [ ... ] . [ digit [ ... ] ] | . digit [ ... ] }
```

Comment:
seems better to mention the fraction literals together:
```
decimal_digits: [ + | - ] { digit [ ... ] . [ digit [ ... ] ] | . digit [ ... ] }
exponent: E [+ | -] digit [ ... ]

decimal literal: decimal_digits | decimal_digits [exponent] 'BD'
double literal: decimal_digits exponent | decimal_digits [exponent] 'D'
```

Reply:
and there is no way to create float type literals (except using functions like `cast` and `float`). Maybe we should look at the SQL standard and support it later.

Reply:
So the default (without postfix) is a double type literal?

Reply:
I think so, you can try `select typeof(any_value)` to verify

Reply:
I changed your syntax a little to have the following:
```
decimal_digits: [ + | - ] { digit [ ... ] . [ digit [ ... ] ] | . digit [ ... ] }
exponent: E [+ | -] digit [ ... ]
decimal literals: { decimal_digits { [ BD ] | [ exponent BD ] } | digit [ ... ] [exponent] BD }
```
decimal literal:
1, ```decimal_digits ``` 
e.g. 12.578, 123.
2, ```decimal_digits BD ``` 
e.g. 12.578BD, 123.BD
3, ```decimal_digits  exponent BD ```
e.g. -.1234567E+2BD, 1.23E-2BD (1.23E-2 is Double)
4, ```digit [ ... ]  BD ``` 
e.g. 20BD 
5, ```digit [ ... ]  exponent BD ``` 
e.g. 20E5BD (20E5 is Double)
```
double literals: { decimal_digits  { D | exponent [ D ] }  | digit [ ... ] { exponent [ D ] | [ exponent ] D }
```
1, ```decimal_digits  D```
e.g. 12.578D, 123.D
2, ```decimal_digits exponent ```
e.g. -.1234567E1, 1.23E-2 
3, ```decimal_digits exponent  D```
e.g. -.1234567E1D 1.23E-2D
4, ```digit [ ... ] exponent```
e.g.20E5
5, ```digit [ ... ] exponent D```
e.g.20E5D
6, 5, ```digit [ ... ] D```
e.g.20D